staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go

```diff
@@ -335,7 +336,7 @@ func validateNonTrivialConverted(t *testing.T, ctc *conversionTestContext) {
 			client := ctc.versionedClient(ns, createVersion.Name)
 
 			fixture := newConversionMultiVersionFixture(ns, name, createVersion.Name)
-			if !*ctc.crd.Spec.PreserveUnknownFields {
+			if createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields == nil || !*createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields {
```

Comment:
not sure about this change. Do we have fixtures that make use of this schema field? Note that we also still have `crd.Spec.PreserveUnknownFields` in v1, settable through v1beta1 conversion.

Reply:
We don't. Check `multiVersionFixture`: it is static and has `PreserveUnknownFields` set to false, i.e. the if always applies. Just do the same here: remove the if and leave the `SetNestedField` with `foo: garbage`.